pkg/services/serviceaccounts/extsvcaccounts/service.go

```diff
@@ -152,6 +161,24 @@ func (esa *ExtSvcAccountsService) SaveExternalService(ctx context.Context, cmd *
 
 	slug := slugify.Slugify(cmd.Name)
 
+	// update the service account's action sets
+	if esa.features.IsEnabled(ctx, featuremgmt.FlagAccessActionSets) && esa.actionSetsSvc != nil {
+		svc := *esa.actionSetsSvc
+		// Add the new action sets
+		if cmd.Self.ActionSets != nil {
+			for _, actionSet := range cmd.Self.ActionSets {
+				// TODO: need to know which resource, permissions and actions are needed
+				svc.StoreActionSet(slug, actionSet.Action, actionSet.Actions)
+				// add the actionset as part of the permissions
+				// convert actionset to permissions
+				// add the permissions to the service account
+				cmd.Self.Permissions = append(cmd.Self.Permissions, ac.Permission{
```

Comment:
this stores the permission into the db.

this could be removed from this PR if needed